server/sandbox_network.go

```diff
@@ -30,13 +30,13 @@ func (s *Server) networkStart(ctx context.Context, sb *sandbox.Sandbox) (podIPs
 
 	podNetwork, err := s.newPodNetwork(sb)
 	if err != nil {
-		return
+		return nil, nil, err
 	}
 
 	// Ensure network resources are cleaned up if the plugin succeeded
 	// but an error happened between plugin success and the end of networkStart()
 	defer func() {
-		if err != nil {
+		if errRet != nil {
```

Comment:
Looks like this code won't compile, as  you forgot to change the function signature above, it still looks this way:

```go
func (s *Server) networkStart(ctx context.Context, sb *sandbox.Sandbox) (podIPs []string, result cnitypes.Result, err error) {
```